data-hibernate-reactive/src/test/java/io/micronaut/data/hibernate/reactive/AuthorRepository.java

```diff
@@ -52,10 +52,14 @@ public interface AuthorRepository extends ReactorCrudRepository<Author, Long> {
     Mono<Author> findByName(String name);
 
     @Join("books")
+    @Join(value = "books.genre", type = Join.Type.LEFT_FETCH)
```

Comment:
So, this was the issue in new hibernate reactive. There are `@OneToOne` Genre mapping and `@ManyToOne` Publisher mapping on the Book entity. Reactive hibernate does not find correct matching `DomainResultAssembler` when initialising an entity. Probably because these mappings are null, not populated or returned from the db, then it finds `ReactiveEntityAssembler` that should handle that entity instead to ignore it. Then this one throws error `nonReactiveMethodCall`.
We can go on with this workaround, at least it's not issue with our code and can keep an eye on hibernate reactive or perhaps report an issue which is reproducible.